### ISSUE TITLE
Transparent HTTPS proxy

### ISSUE THREAD

**Issue by filippocarletti** [NethServer/dev#5169]
Switch squid ssl proxy from bump to peek and splice, squid 3.5 required.
New implementation will have the following advantages:

- No certificate to install on all browsers
- No untrusted certificate warning
- No sniffing on bank account :slight_smile:
- Seamless filter of unwanted websites, both HTTP and HTTPS

Steps:
- [x] require squid 3.5
- [x] handle updates from squid 3.3.8
- [x] refactor squid template
- [x] remove SSL bypass ACL (see also bug on NS 6:  http://dev.nethserver.org/issues/3432)

Details:
http://community.nethserver.org/t/transparent-https-proxy/5064

**Comment by filippocarletti** [comment]
Test case;
1. install the web proxy module on a freshly installed system
2. update a system where the proxy is already in use: the download cert UI section should go away
3. set the proxy Transparent with SSL option, browse to some https website and look into /var/log/squid/access.log.
4. check that the certificate is trusted by browser and belongs to the original website

**Comment by gsanchietti** [comment]
In nethserver testing:
- nethserver-squid-1.4.0-1.5.gf9cf6e6.ns7.noarch.rpm

Before testing, make sure *cr* repository is enabled:
```
yum --enablerepo=cr,nethserver-testing install squid nethserver-squid
```

**Comment by gsanchietti** [comment]
Verified.

Added new documentation on developer manual: https://github.com/NethServer/nethserver-squid#ssl-peek-and-splice

**Comment by gsanchietti** [comment]
Released in 7.3.1611/base:
- nethserver-squid-1.5.0-1.ns7.noarch.rpm
